treeherder/webapp/api/jobs.py

```diff
@@ -142,13 +142,17 @@ def retrigger(self, request, project, jm):
         pulse message.
         """
         job_id_list = request.data["job_id_list"]
+        success = False
         for pk in job_id_list:
             job = jm.get_job(pk)
             if job:
                 jm.retrigger(request.user.email, job[0])
-                return Response({"message": "retriggered job '{0}'".format(job[0]['job_guid'])})
-            else:
-                return Response("No job with id: {0}".format(pk), 404)
+                success = True
```

Comment:
You are not covering the case where not all the jobs were successfully retriggered. It's probably better to invert the logic and keep track of eventual failures, telling the user that at least one job failed to retrigger.